packages/edit-site/src/components/layout/router.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { useIsSiteEditorLoading } from './hooks';
+import Editor from '../editor';
+import DataviewsPatterns from '../page-patterns/dataviews-patterns';
+import PagePages from '../page-pages';
+import PagePatterns from '../page-patterns';
+import PageTemplateParts from '../page-template-parts';
+import PageTemplates from '../page-templates';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+export default function useLayoutAreas() {
+	const isSiteEditorLoading = useIsSiteEditorLoading();
+	const { params } = useLocation();
+	const { postType, postId, path, type, isCustom } = params ?? {};
+
+	// Regular page
+	if ( path === '/page' ) {
+		const isListLayout =
+			isCustom !== 'true' && ( ! type || type === 'list' );
+		return {
+			areas: {
+				content: window.__experimentalAdminViews ? (
+					<PagePages />
+				) : undefined,
+				preview: ( isListLayout ||
+					! window.__experimentalAdminViews ) && (
+					<Editor isLoading={ isSiteEditorLoading } />
+				),
+			},
+			widths: {
+				content:
+					window.__experimentalAdminViews && isListLayout
+						? 380
+						: undefined,
+			},
+		};
+	}
+
+	// Regular other post types
+	if ( postType && postId ) {
```

Comment:
This is the details view, as it exists today. As a follow-up, we'll need to update this to absorb the details as part of the content area (it's now part of the sidebar). ([Mockup](https://github.com/WordPress/gutenberg/pull/57523)).

Reply:
Indeed, we might need to have it specific per post type. Remains to be seen.